### PR TITLE
external pg support 

### PR DESCRIPTION
**Pull request by ozRunAI**

### CODE REVIEW

File: docs/admin/runai-setup/self-hosted/k8s/preparations.md [hunk 1]
```diff
@@ -77,6 +77,7 @@ You must provide the domain's private key and crt as a Kubernetes secret in the
 kubectl create secret tls runai-backend-tls -n runai-backend \
     --cert /path/to/fullchain.pem --key /path/to/private.pem
 ```
+
```
Comment (jasonnovichRunAI) [inline docs/admin/runai-setup/self-hosted/k8s/preparations.md:80]: @ozRunAI no need for this extra empty line.

File: docs/admin/runai-setup/self-hosted/k8s/preparations.md [hunk 2]
```diff
@@ -94,6 +95,57 @@ kubectl label node <NODE-NAME> node-role.kubernetes.io/runai-system=true
...
+    \c <DATABASE_NAME> 
+
+    -- grafana
+    CREATE ROLE grafana WITH LOGIN PASSWORD '<GRAFANA_PASSWORD>'; 
+    CREATE SCHEMA grafana authorization grafana;
+    ALTER USER grafana set search_path='grafana';
+    -- Exit psql
+    \q
+    ```
+
```
Comment (jasonnovichRunAI) [inline docs/admin/runai-setup/self-hosted/k8s/preparations.md:129]: @ozRunAI remove the empty line.

File: docs/admin/runai-setup/self-hosted/k8s/prerequisites.md [hunk 3]
```diff
@@ -66,20 +66,26 @@ The Run:ai control plane installation assumes an existing installation of NGINX
...
+ > The Run:ai control plane, when installed without a Run:ai cluster, does not require the Prometheus prerequisites. 
 
 
 ### Inference (optional)
 
 See Run:ai Cluster prerequisites [Inference](../../cluster-setup/cluster-prerequisites.md#inference) requirements.
 
-The Run:ai control plane, when installed without a Run:ai cluster, does not require the Inference prerequisites. 
+ > The Run:ai control plane, when installed without a Run:ai cluster, does not require the Inference prerequisites. 
+
```
Comment (jasonnovichRunAI) [inline docs/admin/runai-setup/self-hosted/k8s/prerequisites.md:83]: @ozRunAI remove empty line.

File: docs/admin/runai-setup/self-hosted/ocp/backend.md [hunk 4]
```diff
@@ -4,6 +4,7 @@
 
 Make sure you have followed the Control Plane [prerequisites](./prerequisites.md) and [preperations](./preperations.md).
 
+
```
Comment (jasonnovichRunAI) [inline docs/admin/runai-setup/self-hosted/ocp/backend.md:7]: @ozRunAI remove this line

File: docs/admin/runai-setup/self-hosted/ocp/backend.md [hunk 5]
```diff
@@ -61,6 +68,17 @@ There may be cases where you need to set additional properties as follows:
...
+If you have opted to connect to an [external PostgreSQL database](preperations.md#external-postgres-database-optional),  refer to the additional configurations table below. Adjust the following parameters based on your connection details:
+
+* `postgresql.enabled` - set to `false`
+* `global.postgresql.auth.password`
+* `global.postgresql.auth.username`
+* `global.postgresql.auth.host`
+* `global.postgresql.auth.port`
+* `grafana.dbUser`
+* `grafana.dbPassword`
```
Comment (jasonnovichRunAI) [inline docs/admin/runai-setup/self-hosted/ocp/backend.md:82]: @ozRunAI remove empty line

File: docs/admin/runai-setup/self-hosted/ocp/preparations.md [hunk 6]
```diff
@@ -83,6 +83,56 @@ kubectl label node <NODE-NAME> node-role.kubernetes.io/runai-system=true
...
+    \c <DATABASE_NAME> 
+
+    -- grafana
+    CREATE ROLE grafana WITH LOGIN PASSWORD '<GRAFANA_PASSWORD>'; 
+    CREATE SCHEMA grafana authorization grafana;
+    ALTER USER grafana set search_path='grafana';
+    -- Exit psql
+    \q
+    ```
+
```
Comment (jasonnovichRunAI) [inline docs/admin/runai-setup/self-hosted/ocp/preparations.md:117]: @ozRunAI remove empty line

**Review by jasonnovichRunAI — COMMENTED** [review]
@ozRunAI @JamieWeider72 Some basic markdown errors. Also if you can, build out a branch for changes from your local and not make edits in your master. (ozRunAI:master)
Also, it would be best to use a Jira ticket in the commit message so that the branch and commits can be tracked.

File: docs/admin/runai-setup/self-hosted/k8s/backend.md [hunk 7]
```diff
@@ -51,9 +51,16 @@ There may be cases where you need to set additional properties as follows:
 | `global.keycloakx.host` |  KeyCloak (Run:ai internal identity provider) host path | Override the DNS for Keycloak. This can be used to access Keycloak from outside the Run:ai Control Plane cluster via ingress | 
 | `global.ingress.ingressClass` |  Ingress class  |  Run:ai default is using NGINX. If your cluster has a different ingress controller, you can configure the ingress class to be created by Run:ai |
 | `global.ingress.tlsSecretName`  | TLS secret name  | Run:ai requires the creation of a secret with domain certificate. See [above](#domain-certificate). If the `runai-backend` namespace already had such a secret, you can set the secret name here  |
-| `global.postgresql.auth.username`  | PostgreSQL username | Override the Run:ai default user name for the Run:ai database  |
-| `global.postgresql.auth.password`  | PostgreSQL password | Override the Run:ai default password for the Run:ai database  |
-| `global.postgresql.auth.port`  | PostgreSQL port | Override the default PostgreSQL port for the Run:ai database  |
+
```
Comment (yossig-runai) [inline docs/admin/runai-setup/self-hosted/k8s/backend.md:54]: please remove the new line

File: docs/admin/runai-setup/self-hosted/k8s/preparations.md [hunk 8]
```diff
@@ -94,6 +95,57 @@ kubectl label node <NODE-NAME> node-role.kubernetes.io/runai-system=true
 !!! Warning
     Do not select the Kubernetes master as a `runai-system` node. This may cause Kubernetes to stop working (specifically if Kubernetes API Server is configured on 443 instead of the default 6443).
 
+### External Postgres database (optional)
+
+If you have opted to use an [external PostgreSQL database](prerequisites.md#external-postgresql-database-optional).
+, you need to perform initial setup to ensure successful installation. Follow these steps:
+
+1. Create a SQL script file, edit the parameters below, and save it locally:
```
Comment (yossig-runai) [inline docs/admin/runai-setup/self-hosted/k8s/preparations.md:103]: Maybe something like:
1. Create a SQL script file, edit the parameters below, and save it locally:
    * Replace `<DATABASE_NAME>` with a dedicate database name for RunAi  in your PostgreSQL database.
    * Replace `<ROLE_NAME>` with a dedicated  role name (user)  for RunAi database.
    * Replace `<ROLE_PASSWORD>` with a password for the new PostgreSQL role.
    * Replace `<GRAFANA_PASSWORD>` with the password to be set for Grafana integration.

File: docs/admin/runai-setup/self-hosted/ocp/preparations.md [hunk 9]
```diff
@@ -83,6 +83,56 @@ kubectl label node <NODE-NAME> node-role.kubernetes.io/runai-system=true
 !!! Warning
     Do not select the Kubernetes master as a `runai-system` node. This may cause Kubernetes to stop working (specifically if Kubernetes API Server is configured on 443 instead of the default 6443).
 
+### External Postgres database (optional)
+
+If you have opted to use an [external PostgreSQL database](prerequisites.md#external-postgresql-database-optional).
+, you need to perform initial setup to ensure successful installation. Follow these steps:
+
+1. Create a SQL script file, edit the parameters below, and save it locally:
```
Comment (yossig-runai) [inline docs/admin/runai-setup/self-hosted/ocp/preparations.md:91]: Maybe something like:

Create a SQL script file, edit the parameters below, and save it locally:
Replace <DATABASE_NAME> with a dedicate database name for RunAi in your PostgreSQL database.
Replace <ROLE_NAME> with a dedicated role name (user) for RunAi database.
Replace <ROLE_PASSWORD> with a password for the new PostgreSQL role.
Replace <GRAFANA_PASSWORD> with the password to be set for Grafana integration.

**Review by yossig-runai — APPROVED** [review]

File: docs/admin/runai-setup/self-hosted/k8s/preparations.md [hunk 10]
```diff
@@ -94,6 +94,56 @@ kubectl label node <NODE-NAME> node-role.kubernetes.io/runai-system=true
 !!! Warning
     Do not select the Kubernetes master as a `runai-system` node. This may cause Kubernetes to stop working (specifically if Kubernetes API Server is configured on 443 instead of the default 6443).
 
+### External Postgres database (optional)
+
+If you have opted to use an [external PostgreSQL database](prerequisites.md#external-postgresql-database-optional).
+, you need to perform initial setup to ensure successful installation. Follow these steps:
```
Comment (JamieWeider72) [inline docs/admin/runai-setup/self-hosted/k8s/preparations.md:100]:
```suggestion
If you have opted to use an [external PostgreSQL database](prerequisites.md#external-postgresql-database-optional), you need to perform initial setup to ensure successful installation. Follow these steps:
```

File: docs/admin/runai-setup/self-hosted/k8s/backend.md [hunk 11]
```diff
@@ -62,6 +68,18 @@ There may be cases where you need to set additional properties as follows:
 
 Use the `--set` syntax in the helm command above.  
 
+#### External PostgreSQL database
+
+If you have opted to connect to an [external PostgreSQL database](preperations.md#external-postgres-database-optional),  refer to the additional configurations table below. Adjust the following parameters based on your connection details:
```
Comment (JamieWeider72) [inline docs/admin/runai-setup/self-hosted/k8s/backend.md:73]:
```suggestion
If you have opted to connect to an [external PostgreSQL database](preperations.md#external-postgres-database-optional), refer to the additional configurations table below. Adjust the following parameters based on your connection details:
```

File: docs/admin/runai-setup/self-hosted/k8s/preparations.md [hunk 12]
```diff
@@ -94,6 +94,56 @@ kubectl label node <NODE-NAME> node-role.kubernetes.io/runai-system=true
...
+
+    -- grafana
+    CREATE ROLE grafana WITH LOGIN PASSWORD '<GRAFANA_PASSWORD>'; 
+    CREATE SCHEMA grafana authorization grafana;
+    ALTER USER grafana set search_path='grafana';
+    -- Exit psql
+    \q
+    ```
+
+2. run the following command on a machine where PostgreSQL client (`pgsql`) is installed:
```
Comment (JamieWeider72) [inline docs/admin/runai-setup/self-hosted/k8s/preparations.md:129]:
```suggestion
2. Run the following command on a machine where PostgreSQL client (`pgsql`) is installed:
```

File: docs/admin/runai-setup/self-hosted/k8s/prerequisites.md [hunk 13]
```diff
@@ -66,20 +66,25 @@ The Run:ai control plane installation assumes an existing installation of NGINX
...
+### External Postgres database (optional)
+
+The Run:ai Control Plane installation includes a default PostgreSQL database. However, you may opt to use an existing PostgreSQL database if you have specific requirements or preferences. Please ensure that your PostgreSQL database is version 16 or higher.
```
Comment (JamieWeider72) [inline docs/admin/runai-setup/self-hosted/k8s/prerequisites.md:86]:
```suggestion
The Run:ai control plane installation includes a default PostgreSQL database. However, you may opt to use an existing PostgreSQL database if you have specific requirements or preferences. Please ensure that your PostgreSQL database is version 16 or higher.
```

File: docs/admin/runai-setup/self-hosted/ocp/backend.md [hunk 14]
```diff
@@ -61,7 +67,17 @@ There may be cases where you need to set additional properties as follows:
 
 Use the `--set` syntax in the helm command above.  
 
+#### External PostgreSQL database
 
+If you have opted to connect to an [external PostgreSQL database](preperations.md#external-postgres-database-optional),  refer to the additional configurations table below. Adjust the following parameters based on your connection details:
```
Comment (JamieWeider72) [inline docs/admin/runai-setup/self-hosted/ocp/backend.md:72]:
```suggestion
If you have opted to connect to an [external PostgreSQL database](preperations.md#external-postgres-database-optional), refer to the additional configurations table below. Adjust the following parameters based on your connection details:
```

File: docs/admin/runai-setup/self-hosted/ocp/preparations.md [hunk 15]
```diff
@@ -83,6 +83,55 @@ kubectl label node <NODE-NAME> node-role.kubernetes.io/runai-system=true
 !!! Warning
     Do not select the Kubernetes master as a `runai-system` node. This may cause Kubernetes to stop working (specifically if Kubernetes API Server is configured on 443 instead of the default 6443).
 
+### External Postgres database (optional)
+
+If you have opted to use an [external PostgreSQL database](prerequisites.md#external-postgresql-database-optional).
+, you need to perform initial setup to ensure successful installation. Follow these steps:
```
Comment (JamieWeider72) [inline docs/admin/runai-setup/self-hosted/ocp/preparations.md:89]:
```suggestion
If you have opted to use an [external PostgreSQL database](prerequisites.md#external-postgresql-database-optional), you need to perform initial setup to ensure successful installation. Follow these steps:
```

File: docs/admin/runai-setup/self-hosted/ocp/preparations.md [hunk 16]
```diff
@@ -83,6 +83,55 @@ kubectl label node <NODE-NAME> node-role.kubernetes.io/runai-system=true
...
+
+    -- grafana
+    CREATE ROLE grafana WITH LOGIN PASSWORD '<GRAFANA_PASSWORD>'; 
+    CREATE SCHEMA grafana authorization grafana;
+    ALTER USER grafana set search_path='grafana';
+    -- Exit psql
+    \q
+    ```
+
+2. run the following command on a machine where PostgreSQL client (`pgsql`) is installed:
```
Comment (JamieWeider72) [inline docs/admin/runai-setup/self-hosted/ocp/preparations.md:118]:
```suggestion
2. Run the following command on a machine where PostgreSQL client (`pgsql`) is installed:
```

File: docs/admin/runai-setup/self-hosted/ocp/prerequisites.md [hunk 17]
```diff
@@ -58,7 +58,11 @@ Information on how to download the GPU Operator for air-gapped installation can
 
 See Run:ai Cluster prerequisites [Inference](../../cluster-setup/cluster-prerequisites.md#inference) requirements.
 
-The Run:ai control plane, when installed without a Run:ai cluster, does not require the Inference prerequisites. 
+> The Run:ai control plane, when installed without a Run:ai cluster, does not require the Inference prerequisites. 
+
+### External PostgreSQL database (optional)
+
+The Run:ai Control Plane installation includes a default PostgreSQL database. However, you may opt to use an existing PostgreSQL database if you have specific requirements or preferences. Please ensure that your PostgreSQL database is version 16 or higher.
```
Comment (JamieWeider72) [inline docs/admin/runai-setup/self-hosted/ocp/prerequisites.md:65]:
```suggestion
The Run:ai control plane installation includes a default PostgreSQL database. However, you may opt to use an existing PostgreSQL database if you have specific requirements or preferences. Please ensure that your PostgreSQL database is version 16 or higher.
```

**Pull request merged**